Userland/Libraries/LibXML/Parser/Parser.h

```diff
@@ -29,6 +29,7 @@ struct ParseError {
 struct Listener {
     virtual ~Listener() { }
 
+    virtual void set_source(StringView) { }
```

Comment:
This would feel cleaner if we changed `Document::set_source()` to take a `String` (instead of the current `String const&`) and then had one here as well and `move()` it around - this looks suspiciously like someone is going to store a view and UAF that :sweat_smile: